src/test/scala/is/hail/expr/ir/ExtractAggregatorsSuite.scala

```diff
@@ -0,0 +1,243 @@
+package is.hail.expr.ir
+
+import is.hail.asm4s._
+import is.hail.annotations._
+import ScalaToRegionValue._
+import is.hail.annotations.aggregators._
+import is.hail.check._
+import is.hail.expr.{TAggregable, TArray, TFloat64, TInt32, TStruct, RegionValueAggregator, hailType, HailRep, TBoolean, TFloat32, TInt64, Type}
+import is.hail.expr.ir._
+import org.testng.annotations.Test
+import org.scalatest._
+import Matchers._
+
+class ExtractAggregatorsSuite {
+
+  private def defaultValue(t: Type): Any = t match {
+    case _: TBoolean => false
+    case _: TInt32 => 0
+    case _: TInt64 => 0L
+    case _: TFloat32 => 0.0f
+    case _: TFloat64 => 0.0
+    case _ => 0L // reference types
+  }
+
+  private def packageResults(region: MemoryBuffer, t: TStruct, a: Array[RegionValueAggregator]): Long = {
+    val rvb = new RegionValueBuilder()
+    rvb.set(region)
+    rvb.start(t)
+    rvb.startStruct()
+    t.fields.map(_.typ).zip(a).foreach { case (t, agg) =>
+      rvb.addRegionValue(t, region, agg.result(region))
+    }
+    rvb.endStruct()
+    rvb.end()
+  }
+
+  private def loadIRIntermediate(typ: Type): (MemoryBuffer, Long) => Any = typ match {
```

Comment:
I wasn't sure this was appropriate to put on a Region, it should only be used by tests.